src/features/metadata-instance-editor/__tests__/TemplatedInstance-test.js

```diff
@@ -160,4 +162,19 @@ describe('features/metadata-instance-editor/fields/TemplatedInstance', () => {
         );
         expect(wrapper).toMatchSnapshot();
     });
+
+    test('should correctly render fields corresponding to the included field IDs only', () => {
+        const wrapper = shallow(
+            <TemplatedInstance
+                data={data}
+                dataValue="value"
+                errors={{}}
+                includedFieldIds={includedFieldIds}
+                template={{
+                    fields,
+                }}
+            />,
+        );
+        expect(wrapper).toMatchSnapshot();
```

Comment:
We should try to avoid snapshots, especially on such large components. Can we add more explicit test cases?